osquery/config/plugins/filesystem.cpp

```diff
@@ -9,19 +9,28 @@
  */
 
 #include <fstream>
+#include <iostream>
 
 #include <boost/filesystem/operations.hpp>
+#include <boost/property_tree/ptree.hpp>
+#include <boost/property_tree/json_parser.hpp>
 
 #include <osquery/config.h>
 #include <osquery/flags.h>
 #include <osquery/logger.h>
+#include <osquery/filesystem.h>
 
 namespace fs = boost::filesystem;
+namespace pt = boost::property_tree;
 using osquery::Status;
 
 namespace osquery {
 
 FLAG(string, config_path, "/var/osquery/osquery.conf", "Path to config file");
+FLAG(string,
+     config_extra_files,
+     "/var/osquery/osquery.conf.d/%.conf",
```

Comment:
Nix the "%.conf" in the flag (if you even want a flag), it'll be assumed to match anything in that dir using "%.conf".